backends/cadence/aot/fuse_ops.py

```diff
@@ -907,42 +907,28 @@ def can_fuse_for_chain(
         if not super().can_fuse_for_chain(producer, consumer, consumer_op_packets):
             return False
 
-        def get_dims(node: torch.fx.Node) -> tuple[int, int]:
-            def canonicalize(dim: int) -> int:
-                if dim < 0:
-                    dim += len(node.meta["val"].shape)
-                return dim
-
-            return tuple(canonicalize(cast(int, d)) for d in node.args[1:3])
-
-        def is_equivalent(
-            shape: Sequence[int],
-            transpose0: tuple[int, int],
-            transpose1: tuple[int, int],
-        ) -> bool:
-            def permute_order(
-                order: Sequence[int], dims: tuple[int, int]
-            ) -> Sequence[int]:
-                new_order = list(order)
-                new_order[dims[0]], new_order[dims[1]] = (
-                    new_order[dims[1]],
-                    new_order[dims[0]],
-                )
-                return new_order
+        input_shape = list(cast(torch.fx.Node, producer.args[0]).meta["val"].shape)
 
-            order = permute_order(range(len(shape)), transpose0)
-            order = permute_order(order, transpose1)
+        intermediate_shape = (
+            get_transposed_dims(producer, input_shape)
+            if producer.target == exir_ops.edge.aten.transpose_copy.int
+            else get_permuted_dims(producer, input_shape)
+        )
 
-            non_unit_dims = [dim for dim in range(len(shape)) if shape[dim] != 1]
-            non_unit_dims_permuted = [dim for dim in order if shape[dim] != 1]
+        final_shape = (
+            get_transposed_dims(consumer, intermediate_shape)
+            if consumer.target == exir_ops.edge.aten.transpose_copy.int
+            else get_permuted_dims(consumer, intermediate_shape)
+        )
 
-            return non_unit_dims == non_unit_dims_permuted
+        non_unit_dims = [
+            input_shape[dim] for dim in range(len(input_shape)) if input_shape[dim] != 1
+        ]
+        non_unit_dims_permuted = [
+            final_shape[dim] for dim in range(len(final_shape)) if final_shape[dim] != 1
+        ]
 
-        return is_equivalent(
-            cast(torch.fx.Node, producer.args[0]).meta["val"].shape,
-            get_dims(producer),
-            get_dims(consumer),
-        )
+        return non_unit_dims == non_unit_dims_permuted
```

Comment:
it seems that we are checking that the combination of 2 permutations is an identity based on the start and final size of the shapes, instead of on the permutation itself
(so if the input shape is [4 , 4, 4, 4] then this will always trigger)